astropy/io/fits/connect.py

```diff
@@ -272,30 +286,25 @@ def read_table_fits(
         # preserve null values.
         masked = mask = False
         fill_value = None
-        coltype = col.dtype.subdtype[0].type if col.dtype.subdtype else col.dtype.type
         if col.null is not None:
-            mask = data[col.name] == col.null
+            mask = arr == col.null
```

Comment:
I'm confused how the scaling affects this, but clearly it was the same before, so should be fine.

Reply:
yes here it should not matter to compute the mask.